BackendBench/__init__.py

```diff
@@ -8,4 +8,110 @@
 BackendBench: A PyTorch backend evaluation framework.
 """
 
+import torch
+import os
+import warnings
+from typing import Optional, Union
+from pathlib import Path
+
+# Import the existing DirectoryBackend implementation
+from BackendBench.backends.directory import DirectoryBackend
+
+
 __version__ = "0.1.0"
+__all__ = ["enable", "disable"]
+
+# Global state
+_lib = None
+
+
+def _monkey_patch_operators(op_custom_impl):
+    """
+    Replace PyTorch operators with custom implementations using torch.library.
+    """
+    global _lib
+
+    # Use torch.library to register custom implementations
+    _lib = torch.library.Library("aten", "IMPL", "CPU")
+
+    patched_count = 0
+    for op, custom_impl in op_custom_impl.items():
+        try:
+            # Extract operator name and overload from the OpOverload
+            op_name = op._schema.name
+            overload_name = op._schema.overload_name
+
+            # Create the full name for torch.library
+            if overload_name:
+                full_name = f"{op_name}.{overload_name}"
+            else:
+                full_name = op_name
+
+            # Register the custom implementation
+            _lib.impl(full_name, custom_impl, "CPU")
+            patched_count += 1
+
+        except Exception as e:
+            # Some operators might not be patchable
+            warnings.warn(f"Could not register {op}: {e}")
+
+    if patched_count > 0:
+        print(f"Successfully registered {patched_count} custom operators")
+
+
+def enable(
+    kernel_dir: Optional[Union[str, Path]] = None,
+    verbose: bool = False,
+) -> None:
+    """
+    Enable the DirectoryBackend to use custom operator implementations.
+    """
+    global _lib
+
+    if _lib is not None:
+        warnings.warn("DirectoryBackend is already enabled. Call disable() first.")
+        return
+
+    # Set default kernel directory
+    if kernel_dir is None:
+        kernel_dir = os.path.join(os.getcwd(), "generated_kernels")
```

Comment:
not a huge fan of cwd just because the behavior will be different depending on where i pip isntall from and I prefer doing things relative to `current_file = Path(__file__)`